projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -992,7 +1079,49 @@ public ValueTask BasicPublishAsync<TProperties>(CachedString exchange, CachedStr
             }
 
             var cmd = new BasicPublishMemory(exchange.Bytes, routingKey.Bytes, mandatory, default);
-            return ModelSendAsync(in cmd, in basicProperties, body);
+            RabbitMQActivitySource.TryGetExistingContext(basicProperties, out ActivityContext existingContext);
+            using Activity sendActivity = RabbitMQActivitySource.PublisherHasListeners
+                ? RabbitMQActivitySource.Send(routingKey.Value, exchange.Value, body.Length, existingContext)
+                : default;
+
+            if (sendActivity != null)
+            {
+                BasicProperties props = PopulateActivityAndPropagateTraceId(basicProperties, sendActivity);
+                await ModelSendAsync(in cmd, in props, body);
+                return;
+            }
+
+            await ModelSendAsync(in cmd, in basicProperties, body);
+        }
+
+        private static BasicProperties PopulateActivityAndPropagateTraceId<TProperties>(TProperties basicProperties,
+            Activity sendActivity) where TProperties : IReadOnlyBasicProperties, IAmqpHeader
+        {
+            // This activity is marked as recorded, so let's propagate the trace and span ids.
+            if (sendActivity.IsAllDataRequested)
+            {
+                if (!string.IsNullOrEmpty(basicProperties.CorrelationId))
+                {
+                    sendActivity.SetTag(RabbitMQActivitySource.MessageConversationId, basicProperties.CorrelationId);
+                }
+            }
+
+            BasicProperties props = default;
+            if (basicProperties is BasicProperties properties)
+            {
+                props = properties;
+            }
+            else if (basicProperties is EmptyBasicProperty)
+            {
+                props = new BasicProperties();
+            }
+
+            var headers = props.Headers ?? new Dictionary<string, object>();
+
+            // Inject the ActivityContext into the message headers to propagate trace context to the receiving service.
+            DistributedContextPropagator.Current.Inject(sendActivity, headers, InjectTraceContextIntoBasicProperties);
```

Comment:
This could have some confusing outcomes. Since OpenTelemetry Baggage doesn't propagate Activity.Baggage, if they're using OpenTelemetry to populate things, then this won't have the desired effect.

You could either take a dependency on OpenTelemetry.Api, or make it clear that Baggage isn't included.

Reply:
Its best to document this and unblock this PR! 
Solving the 2 different baggages is something OTel .NET and .NET runtime needs to collaborate and fix.

Reply:
I think the problem here is that the user either has to sacrifice baggage functionality, or reimplement and ignore all the instrumentation in this PR.

What might be better here is potentially add a hook so that a custom inject/extract method can be added in it's place, with the default having already been set using the `DistributedContextPropagator`. That way, the OpenTelemetry extension, that will need to have the Otel dependency, can add this.

I think that's probably a more elegant way to handle this.

Reply:
The way I see it right now, this might actually do harm without the baggage integration.

There's an expectation that baggage will be propagated, and turning this on saying "propagation works" will be misleading.

I get that .NET doesn't support baggage properly for valid reasons, but the main usecase for this is to enable OpenTelemetry and that would be misleading as it doesn't fully support opentelemetry.

Reply:
@michaelklishin @lukebakken I'll defer the decision to you on whether we postpone the "proper" baggage handling (which already differs for example in the Azure SDK as per the comment from @lmolkova) to a later PR and and document the differences or if we should do this properly and add more to this PR. In the meantime I'm going to rebase this PR on the latest changes.

Reply:
Thanks for doing the rebase. To be honest, I have only the most superficial understanding of OpenTelemetry and am depending on this PR's participants to ensure a correct implementation.